src/main/java/site/coach_coach/coach_coach_server/routine/service/RoutineService.java

```diff
@@ -99,4 +103,27 @@ public UserInfoForRoutineList getUserInfoForRoutineList(Long userIdParam, Long c
 				userInfo.getProfileImageUrl());
 		}
 	}
+
+	public void createRoutine(CreateRoutineRequest createRoutineRequest, Long userIdByJwt) {
+		Sport sportInfo = Sport.builder()
+			.sportId(createRoutineRequest.sportId())
+			.build();
+
+		if (createRoutineRequest.userId() == null) {
+			Routine newRoutine = Routine.builder()
+				.userId(userIdByJwt)
+				.routineName(createRoutineRequest.routineName())
+				.sport(sportInfo)
+				.build();
+			routineRepository.save(newRoutine);
+		} else {
+			Routine newRoutine = Routine.builder()
+				.userId(createRoutineRequest.userId())
+				.coachId(userIdByJwt)
+				.routineName(createRoutineRequest.routineName())
+				.sport(sportInfo)
+				.build();
+			routineRepository.save(newRoutine);
+		}
```

Comment:
공통된 부분이 있는 것 같아서요! 이걸 한 번에 처리하면 더 좋은 코드가 될 것 같아요!!!
```suggestion
    Routine.RoutineBuilder routineBuilder = Routine.builder()
        .userId(createRoutineRequest.userId() == null ? userIdByJwt : createRoutineRequest.userId())
        .routineName(createRoutineRequest.routineName())
        .sport(sportInfo);

    if (createRoutineRequest.userId() != null) {
        routineBuilder.coachId(userIdByJwt);
    }

    routineRepository.save(routineBuilder.build());
```

일단 이것은 그냥 제 생각대로 짠 거라 더 좋은 코드가 있을 수도 있고 예외 사항이 있을 수 있어요!!
그냥 참고해주시면 좋을 것 같아요!

Reply:
저도 Routine.build 과정을 좀 합칠 수 있을것 같다고 생각했는데 Builder를 쓰면 되는군요!
해당 코드에서 좀 더 보완해서 한번 리팩토링 해보겠습니다 ㅎㅎ